### PR TITLE
Refactor Mijia crate a bit

### PR DESCRIPTION
**Pull request by qwandor-google**
Just moving things around and adding a few doc comments.

### CODE REVIEW

File: publish-mqtt/src/main.rs [hunk 1]
```diff
@@ -483,3 +482,23 @@ async fn handle_bluetooth_event(
 
     Ok(())
 }
+
+/// Read the given file of key-value pairs into a hashmap.
+/// Returns an empty hashmap if the file doesn't exist, or an error if it is malformed.
+pub fn hashmap_from_file(filename: &str) -> Result<HashMap<String, String>, io::Error> {
```
Comment (alsuren) [inline publish-mqtt/src/main.rs:488]: (observation) I probably would have put it a bit closer to `run_sensor_system()` (it's the first thing that run_sensor_system() does, so probably immediately below) but in the long run we will probably split this crate out into modules too, so it doesn't really matter where it goes for now.

File: mijia/src/decode.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            self.temperature, self.humidity, self.battery_voltage, self.battery_percent
+        )
+    }
+}
+
+impl Readings {
+    /// Decode the readings from the raw bytes of the Bluetooth characteristic value, if they are
+    /// valid.
+    /// Returns `None` if the value is not valid.
+    pub fn decode(value: &[u8]) -> Option<Readings> {
```
Comment (alsuren) [inline mijia/src/decode.rs:32]: long-term, I would be tempted to implement https://doc.rust-lang.org/std/convert/trait.TryFrom.html , and then use `let readings = value.try_into().ok()?` in the one place where it's used.

For now, I don't think that consumers will ever have `&[u8]` in their hand, so maybe `pub(crate)` and leave everything else as-is?

**Review by alsuren — APPROVED** [review]

File: publish-mqtt/src/main.rs (hunk 1 above)
Comment (qwandor-google) [reply publish-mqtt/src/main.rs]: Done.

File: mijia/src/decode.rs (hunk 2 above)
Comment (qwandor-google) [reply mijia/src/decode.rs]: Done the latter.

**Pull request merged**